scripts/tools/RecompileItemsElements.php

```diff
@@ -0,0 +1,217 @@
+ *
+ * Copyright (c) 2017 (original work) Open Assessment Technologies SA;
+ */
+ 
+namespace oat\taoQtiTest\scripts\tools;
+
+use oat\generis\model\OntologyAwareTrait;
+use \common_report_Report as Report;
+use oat\taoDelivery\model\AssignmentAggregator\UnionAssignmentService;
+use oat\taoDeliveryRdf\model\DeliveryAssemblyService;
+use oat\taoQtiItem\model\QtiJsonItemCompiler;
+use qtism\data\AssessmentItemRef;
+use oat\oatbox\extension\script\ScriptAction;
+
+/**
+ * Class RecompileItemsElements
+ *
+ * Update every deliveries in order to add the metadata of AssessmentItemRef's by AssessmentItemRef Identifier.
+ *
+ * php index.php 'oat\taoQtiTest\scripts\tools\RecompileItemsElements'
+ *
+ * Usage example:
+ * ```
+ * sudo -u www-data php index.php '\oat\taoQtiTest\scripts\tools\RecompileItemsElements'
+ * ```
+ *
+ * @package oat\taoQtiTest\scripts\tools
+ */
+class RecompileItemsElements extends ScriptAction
+{
+    use OntologyAwareTrait;
+
+    /** @var Report */
+    private $report;
+
+    private $wetRun;
+
+    /**
+     * @return string
+     */
+    protected function provideDescription()
+    {
+        return 'Update every deliveries in order to add the metadata of AssessmentItemRef\'s by AssessmentItemRef Identifier.';
+    }
+
+    /**
+     * @return array
+     */
+    protected function provideOptions()
+    {
+        return [
+            'wetRun' => [
+                'longPrefix' => 'wetRun',
+                'required' => false,
+                'description' => 'Wet run',
+                'defaultValue' => 0
+            ]
+        ];
+    }
+
+    /**
+     * @return Report
+     * @throws \common_Exception
+     * @throws \common_exception_Error
+     */
+    protected function run()
+    {
+        try {
+            $this->init();
+        } catch (\Exception $e) {
+            return new Report(Report::TYPE_ERROR, $e->getMessage());
+        }
+        $this->recompileItems();
+        return $this->report;
+    }
+
+    /**
+     * Initialize parameters
+     */
+    protected function init()
+    {
+        $this->wetRun = (boolean) $this->getOption('wetRun');
+        $this->report = new Report(Report::TYPE_INFO, 'Starting recompile deliveries');
+    }
+
+    /**
+     * Recompile items
+     *
+     * @throws \common_Exception
+     * @throws \common_exception_Error
+     */
+    protected function recompileItems()
+    {
+        $extManager = $this->getServiceLocator()->get(\common_ext_ExtensionsManager::SERVICE_ID);
+
+        if ($extManager->isInstalled('taoDeliveryRdf') === true && $extManager->isEnabled('taoDeliveryRdf') === true) {
+            $compiledDeliveryClass = $this->getClass(DeliveryAssemblyService::CLASS_URI);
+            if ($compiledDeliveryClass->exists() === true) {
+                foreach ($compiledDeliveryClass->getInstances(true) as $compiledDelivery) {
+                    $this->getAssessmentsFromDelivery($compiledDelivery);
```

Comment:
You will have to add a catch of exception here in case the delivery it's not valid compile to not stop the execution. I get this exception in the described
```
An exception occured while running "\oat\taoQtiTest\scripts\tools\RecompileItemsElements"
  Property (http://www.tao.lu/Ontologies/TAODelivery.rdf#AssembledDeliveryRuntime) of resource (https://nccersso.taocloud.org/nccer.rdf#i15393348749718555) should not be empty

```